pyaedt/edb_core/materials.py

```diff
@@ -856,116 +845,133 @@ def add_material_from_aedt(self, material_name):
-
-        amat_sys = [
-            os.path.join(dirpath, filename)
-            for dirpath, _, filenames in os.walk(self.syslib)
-            for filename in filenames
-            if fnmatch.fnmatch(filename, "*.amat")
+    def read_materials(mat_file):
+        """Read materials from an AMAT file.
+
+        Parameters
+        ----------
+        filename : str
+            Name of the AMAT file.
+
+        Returns
+        -------
+        dict
+            {material name: dict of material property to value}.
+        """
+
+        def get_line_float_value(line):
+            try:
+                return float(re.split(",|=", line)[-1].strip(")'"))
+            except ValueError:
+                return None
+
+        res = {}
+        _begin_search = re.compile(r"^\$begin '(.+)'")
+        _end_search = re.compile(r"^\$end '(.+)'")
+        amat_keys = [
+            "thermal_conductivity",
+            "permittivity",
+            "dielectric_loss_tangent",
+            "permeability",
+            "magnetic_loss_tangent",
+            "thermal_expansion_coeffcient",
+            "specific_heat",
+            "mass_density",
         ]
-        amat_personal = []
-        if self.personallib:
-            amat_personal = [
-                os.path.join(dirpath, filename)
-                for dirpath, _, filenames in os.walk(self.personallib)
-                for filename in filenames
-                if fnmatch.fnmatch(filename, "*.amat")
-            ]
-        materials = {}
-        for amat in amat_sys:
-            get_mat_list(amat, materials)
-
-        if amat_personal:
-            for amat in amat_personal:
-                get_mat_list(amat, materials)
-        return materials
+        keys = [
+            "thermal_conductivity",
+            "permittivity",
+            "tangent_delta",
+            "permeability",
+            "magnetic_loss_tangent",
+            "thermal_expansion_coeffcient",
+            "specific_heat",
+            "mass_density",
+        ]
+
+        with open(mat_file, "r") as amat_fh:
+            raw_lines = amat_fh.read().splitlines()
+            mat_found = ""
+            for line in raw_lines:
+                b = _begin_search.search(line)
+                if b:  # walk down a level
+                    mat_found = b.group(1)
+                    res.setdefault(mat_found, {})
+                if mat_found:
+                    for amat_key, key in zip(amat_keys, keys):
+                        if amat_key in line:
+                            value = get_line_float_value(line)
+                            if value is not None:
+                                res[mat_found][key] = value
+                    # Extra case to avoid confusion ("conductivity" is included in "thermal_conductivity")
+                    if "conductivity" in line and "thermal_conductivity" not in line:
+                        value = get_line_float_value(line)
+                        if value is not None:
```

Comment:
What if the value is `None`, we just won't assign any value for the conductivity?
I am assuming, the field will stay empty. Just to double check.

Reply:
From my understanding of the previous implementation, we only retrieve what is present in the AMAT file.
If no such value is provided then the field does not exist in the result.
Note that, when one wants to add a new material through `add_material` the method provides a default value for some properties:
- permittivity=1.0006
- permeability=1.0000004
- conductivity=0
- dielectric_loss_tangent=0
- magnetic_loss_tangent=0